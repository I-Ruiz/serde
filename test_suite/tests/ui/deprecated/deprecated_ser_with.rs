#![deny(deprecated)]

use serde::Serializer;
use serde_derive::Serialize;

#[derive(Serialize)]
pub struct Struct {
    #[serde(serialize_with = "deprecated_with")]
    pub field: i32,
}

#[deprecated]
fn deprecated_with<S>(_field: &i32, _serializer: S) -> Result<S::Ok, S::Error>
where
    S: Serializer,
{
    unimplemented!()
}

fn main() {}
