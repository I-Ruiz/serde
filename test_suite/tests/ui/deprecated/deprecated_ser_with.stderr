error: use of deprecated function `deprecated_with`
 --> tests/ui/deprecated/deprecated_ser_with.rs:8:30
  |
8 |     #[serde(serialize_with = "deprecated_with")]
  |                              ^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> tests/ui/deprecated/deprecated_ser_with.rs:1:9
  |
1 | #![deny(deprecated)]
  |         ^^^^^^^^^^
